uniswap-v2/substreams.yaml

```diff
@@ -1,101 +1,241 @@
 package:
-  name: "substreams_uniswap_v2"
+  name: substreams_uniswap_v2
   version: v0.1.0
 
 imports:
   eth: https://github.com/streamingfast/sf-ethereum/releases/download/v0.10.2/ethereum-v0.10.4.spkg
+  entities_change: https://github.com/streamingfast/substreams-entity-change/releases/download/v0.2.0/substreams-entity-change-v0.2.0.spkg
+
+binaries:
+  default:
+    type: wasm/rust-v1
+    file: "../target/wasm32-unknown-unknown/release/substreams_uniswap_v2.wasm"
 
 protobuf:
   files:
     - common.proto
-    - dex_amm.proto
-    - uniswap.proto
     - erc20.proto
+    - uniswap.proto
   importPaths:
-    - ../common/proto
     - ./proto/v1
-    - ./proto/v2
-
-binaries:
-  default:
-    type: wasm/rust-v1
-    file: ../target/wasm32-unknown-unknown/release/substreams_uniswap_v2.wasm
+    - ../common/proto
 
 modules:
-  - name: map_pair_created_events
+  - name: map_pool_created
     kind: map
-    initialBlock: 10000835
+    initialBlock: 10008300
     inputs:
       - source: sf.ethereum.type.v2.Block
     output:
-      type: proto:messari.uniswap.v2.PairCreatedEvents
+      type: proto:messari.uniswap.v2.Pools
 
-  - name: store_pair_created_events
+  - name: store_pools
     kind: store
-    updatePolicy: set
-    valueType: proto:messari.uniswap.v2.PairCreatedEvents
+    updatePolicy: set_if_not_exists
+    valueType: proto:messari.uniswap.v2.Pool
+    inputs:
+      - map: map_pool_created
+
+  - name: store_output_token_supply
+    kind: store
+    updatePolicy: add
+    valueType: bigint
     inputs:
-      - map: map_pair_created_events
+      - source: sf.ethereum.type.v2.Block
+      - store: store_pools
 
-  - name: map_pools
+  - name: map_pool_events
     kind: map
+    initialBlock: 10008300
     inputs:
-      - map: map_pair_created_events
+      - source: sf.ethereum.type.v2.Block
+      - store: store_pools
+      - store: store_output_token_supply
+        mode: deltas
     output:
-      type: proto:messari.dex_amm.v1.Pools
+      type: proto:messari.uniswap.v2.Events
 
-  - name: store_pools
+  - name: store_input_token_balances
     kind: store
     updatePolicy: set
-    valueType: proto:messari.dex_amm.v1.Pools
+    valueType: bigint
     inputs:
-      - map: map_pools
+      - map: map_pool_events
 
-  - name: map_mint_events
-    kind: map
-    initialBlock: 10000835
+  - name: store_native_prices
+    kind: store
+    updatePolicy: set
+    valueType: bigdecimal
     inputs:
-      - source: sf.ethereum.type.v2.Block
       - store: store_pools
-    output:
-      type: proto:messari.uniswap.v2.MintEvents
+      - store: store_input_token_balances
+      - store: store_input_token_balances
+        mode: deltas
+
+  - name: store_usd_prices
+    kind: store
+    updatePolicy: set
+    valueType: bigdecimal
+    inputs:
+      - store: store_pools
+      - map: map_pool_events
+      - store: store_native_prices
+
+  - name: store_pool_tvl
+    kind: store
+    updatePolicy: set
+    valueType: bigdecimal
```

Comment:
So there are a few cases of modules that are like this one. This module for example could be combined with the module above as they are both performing the same store operation without any temporal based conflicts. If they got run together as one module you would only need 1 store squashing event rather than 2 so would save a fair bit of processing if it could be done this way. Two ways to go about this:

1. Change the above module name to be inclusive of this module as well with a name like store_usd_prices_and_pool_tvl and put the fns in the same file.

2. Similar to above but keep store_pool_tvl.rs file in modules folder and don't derive the store macro for it. Instead pass it the store and relevant args from the store_usd_prices module and we can "pretend" it is it's own module even though it's actually just an fn now. (Should help keep each module relatively "clean" even when a lot is going on)